### PR TITLE
Clean up deprecations

### PR DESCRIPTION
**Pull request by alexfikl**
This cleans up some deprecations and makes sure the examples still run.

### CODE REVIEW

File: codepy/libraries.py [hunk 1]
```diff
@@ -123,18 +123,18 @@ def get_boost_libname(basename: str, aksetup: Config) -> list[str]:
...
             getlist(aksetup, "BOOST_INC_DIR", []),
             getlist(aksetup, "BOOST_LIB_DIR", []),
             [
-                *get_boost_libname("python-py{}{}".format(*sys.version_info[:2]),
-                                   aksetup),
-                "python{}.{}{}".format(
-                    *sys.version_info[:2],
-                    "m" if sys.version_info[0] >= 3 else "")
+                *get_boost_libname("python{}{}".format(*version), aksetup),
+                "python{}.{}{}".format(*version, sys.abiflags),
```
Comment (alexfikl) [inline codepy/libraries.py:137]: I'm not sure this is correct. At least on my system
* The boost library is is called `libboost_python313.so`. It wasn't finding the previous one at all.
* Same for `libpython3.13.so`. As far as I can tell, the *m* flag is no longer there: https://docs.python.org/3/library/sys.html#sys.abiflags. I don't know if `abiflags` contains the `t` flag for the free-threaded version..

This seems to match my conda environment on `koelsch`, so it should be a bit more modern.

File: examples/nvcc-test.py [hunk 2]
```diff
@@ -72,55 +73,46 @@
...
 # This is a strange way to create a GPUArray, but is meant to illustrate
 # how to construct a GPUArray if the GPU buffer it owns has been
 # created by something else
 
+length = 25
+constant_value = 2
+
 pointer = pycuda.driver.mem_alloc(length * 4)
 pycuda.driver.memset_d32(pointer, constant_value, length)
 a = pycuda.gpuarray.GPUArray((length,), np.int32, gpudata=pointer)
```
Comment (alexfikl) [inline examples/nvcc-test.py:118]: This example doesn't work for me. It dies somewhere around here when calling `module.adjacentDifference(a)` with
```
TypeError: No registered converter was able to produce a C++ rvalue of type unsigned long long from this Python object of type DeviceAllocation
```

Not quite sure if that's some change in `pycuda` that wasn't updated here or what :(

File: codepy/bpl.py [hunk 3]
```diff
@@ -166,7 +168,8 @@ def compile(self, toolchain, **kwargs):
         """
 
         from codepy.libraries import add_boost_python
-        toolchain = toolchain.copy()
+
+        toolchain = replace(toolchain)
```
Comment (alexfikl) [inline codepy/bpl.py:172]: Also not sure if this replace does what it should. All libraries, includes, etc, in `toolchain` are lists and this just performs a shallow copy..

File: codepy/libraries.py (hunk 1 above)
Comment (inducer) [reply codepy/libraries.py:137]: Not sure there's a "correct" here. Debian used to do its own thing here, but I think they're now consistent with everyone else. "Works for someone" is good enough for me.

**Comment by inducer** [comment]
Thx!

**Pull request merged**